### PR TITLE
Fix menu updates in the multiwin example.

### LINKED ISSUE #728: ui inconsistency in multiwin example
afbeelding

When the menu count is reduced to 0, one of the windows is still showing `custom` menu.

### PR DESCRIPTION
**Pull request by kindlychung**

### CODE REVIEW

File: druid/examples/multiwin.rs [hunk 1]
```diff
@@ -171,39 +171,47 @@ impl AppDelegate<State> for Delegate {
...
             // wouldn't it be nice if a menu (like a button) could just mutate state
             // directly if desired?
-            (Target::Window(id), &MENU_INCREMENT_ACTION) => {
+            &MENU_INCREMENT_ACTION => {
                 data.menu_count += 1;
                 let menu = make_menu::<State>(data);
                 let cmd = Command::new(druid::commands::SET_MENU, menu);
-                ctx.submit_command(cmd, id);
+                for id in &self.windows {
+                    dbg!(id);
```
Comment (jneem) [inline druid/examples/multiwin.rs:199]: Can you remove the dbg printing?

File: druid/examples/multiwin.rs [hunk 2]
```diff
@@ -171,39 +171,47 @@ impl AppDelegate<State> for Delegate {
...
                 false
             }
-            (Target::Window(id), &MENU_DECREMENT_ACTION) => {
+            &MENU_DECREMENT_ACTION => {
                 data.menu_count = data.menu_count.saturating_sub(1);
                 let menu = make_menu::<State>(data);
                 let cmd = Command::new(druid::commands::SET_MENU, menu);
-                ctx.submit_command(cmd, id);
+                for id in &self.windows {
+                    dbg!(id);
```
Comment (jneem) [inline druid/examples/multiwin.rs:209]: Here too

File: druid/examples/multiwin.rs [hunk 3]
```diff
@@ -66,14 +68,10 @@ fn ui_builder() -> impl Widget<State> {
...
-    let inc_button = Button::<State>::new("Add menu item").on_click(|ctx, data, _env| {
-        data.menu_count += 1;
-        ctx.set_menu(make_menu::<State>(data));
-    });
-    let dec_button = Button::<State>::new("Remove menu item").on_click(|ctx, data, _env| {
-        data.menu_count = data.menu_count.saturating_sub(1);
-        ctx.set_menu(make_menu::<State>(data));
-    });
+    let inc_button = Button::<State>::new("Add menu item")
+        .on_click(|ctx, data, _env| ctx.submit_command(MENU_INCREMENT_ACTION, ctx.window_id()));
```
Comment (jneem) [inline druid/examples/multiwin.rs:72]: Do you need the target here? It looks to me like you're handling all the commands globally in the AppDelegate and ignoring the target anyway...

Comment (xStrom) [reply druid/examples/multiwin.rs:72]: The target should probably be `Global` to match the intent.

**Comment by xStrom** [comment]
This needs a rebase with `master` as the `multiwin` example has changed.

**Comment by kindlychung** [comment]
@xStrom plz have another look.

**Comment by kindlychung** [comment]
Done.

**Comment by xStrom** [comment]
You should also add the link definition for the PR number to `CHANGELOG.md` at the bottom where all the others are.

Also rebase with `master` to get CI to pass.

**Comment by kindlychung** [comment]
Done.

**Pull request merged**